equipment_handler/source/adc_handler.c

```diff
@@ -0,0 +1,379 @@
+/*
+ * Copyright (C) 2015  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/**
+ * @file    adc_handler.c
+ * @author  Vasu Gupta
+ * @date    2020-06-15
+ */
+
+#include <ex2_hal/ex2_hyperion_solar_panel_software/equipment_handler/include/adc_handler.h>
+#include <stdint.h>
+// #include <stdio.h>
+//#include "printf.h"
+
+/**
+ * @brief
+ * 		Initialize ADC_Handler for config 1 (include Port, Port Deployable, Starboard, StarBoard Deployable, Zenith)
+ * @details
+ * 		
+ * @return
+ * 		1 == success
+ */
+unsigned char adc_init(uint8_t slave_addr, uint8_t channel) {
+    int delay;
+    // adc_set_command_reg(1<<7,0,0,1,0,1); // this is hardcoded
+    adc_set_command_reg(slave_addr, channel, 0, 1, 1, 0, 1);
+    for(delay=0;delay<10000000;delay++);
+    uint8_t reg_sel = 1; // select read register
+    adc_set_register_pointer(slave_addr, reg_sel);
+    for(delay=0;delay<1000000;delay++);
+
+    // unsigned short data = 0;
+    // unsigned char ch = 0; // channel
+    // adc_get_raw(&data, &ch);
+    // printf("Data: %x, ch: %x\r\n", ch, data);
+
+    return 1;
+}
+
+void adc_write(uint8_t *buf, int size, uint8_t slave_addr) {
+    i2cSetSlaveAdd(ADC_i2c_PORT, slave_addr);
+    i2cSetDirection(i2cREG1, I2C_TRANSMITTER);
+    i2cSetBaudrate(ADC_i2c_PORT, 400);
+    i2cSetCount(ADC_i2c_PORT, size);
+    /* Set mode as Master */
+    i2cSetMode(i2cREG1, I2C_MASTER);
+
+
+    i2cSetStop(ADC_i2c_PORT);
+    /* Transmit Start Condition */
+    i2cSetStart(ADC_i2c_PORT);
+    i2cSend(ADC_i2c_PORT, size, buf);
+
+    /* Wait until Bus Busy is cleared */
+    while(i2cIsBusBusy(ADC_i2c_PORT) == true);
+    i2cSetStop(ADC_i2c_PORT);
+    /* Wait until Stop is detected */
+    //while(i2cIsStopDetected(ADC_i2c_PORT) == 0);
+    /* Clear the Stop condition */
+    i2cClearSCD(ADC_i2c_PORT);
+}
+
+void adc_request_from(uint16_t slave_addr) {
+
+}
```

Comment:
Empty functions?

Reply:
it was previously written by someone else. I'm not sure if we still need it so I just left it there.

Reply:
If possible I'd say do a full repo search for references to it and if nothing comes up then you're safe to delete it. Same for the empty function a few lines down. otherwise just put comments above them explaining their existence

Reply:
It seems there is no references at all. I deleted them